atomix/cluster/src/main/java/io/atomix/raft/snapshot/impl/SnapshotChunkUtil.java

```diff
@@ -23,6 +25,36 @@ static long createChecksum(final byte[] content) {
     return crc32.getValue();
   }
 
+  /**
+   * Returns a new snapshot chunk from a pre 0.24.x install request. Should be removed once no
+   * versions below 0.24.x are supported.
+   *
+   * <p>The snapshot chunk will take most of its information from the install request, and use the
+   * data as the chunk content, while setting the total count to {@link Integer#MIN_VALUE} and the
+   * snapshot checksum to {@link Long#MIN_VALUE} so that neither will be validated against. See
+   * {@link FileBasedReceivedSnapshot} for more on that.
+   *
+   * @param request the install request to build from
+   * @return a valid {@link SnapshotChunk} built from the request
+   */
+  public static SnapshotChunk fromOldInstallRequest(final InstallRequest request) {
```

Comment:
I think we need a mechanism to track these kind of methods or logic, which we expect to remove at some point. Maybe via annotation or adding it to a separate class which can be tracked more easily.

Reply:
I'll group all that legacy stuff together :+1: